stackslib/src/net/http/request.rs

```diff
@@ -34,6 +35,56 @@ use crate::net::http::{
     HttpResponsePreamble, HttpVersion,
 };
 
+/// An HTTP bearer authorization struct
+pub struct HttpBearerAuthorization {
+    /// The bearer authorization token
+    token: String,
+}
+
+impl HttpBearerAuthorization {
+    /// Create a new bearer authorization
+    pub fn new(password: String) -> Self {
+        let token = to_hex(Sha256Sum::from_data(password.as_bytes()).as_bytes());
```

Comment:
I think the main purpose of using hashed tokens is so that you can have the hashed password on disk (in something like an nginx config) without compromising it. On the other hand, we can't do that in the signer, because the signer needs it in plaintext. 

I'm totally fine with keeping this scheme, but the more that I think about it, the more I think it's easier (and the same security) to just check the plain, non-hashed string. This is common with things like API keys, and if think we want more security (I think we don't), we could do an HMAC or signature scheme.

Reply:
Also, this isn't the right way to do HMAC.  You don't want to use SHA256 for it, because it's vulnerable to a length extension attack.

Can we just use a plain password?  Also, do we need a whole struct for this?  Can the block-proposal endpoint just check for the `Authorization` header directly?